From 65476365deac208d8827696f58257b2db57ffeda Mon Sep 17 00:00:00 2001
From: Ivan Vecera <ivecera@redhat.com>
Date: Wed, 2 Jul 2014 13:17:41 -0400
Subject: [ethernet] be2net: enable interrupts in EEH resume

Message-id: <1404307061-5541-1-git-send-email-ivecera@redhat.com>
Patchwork-id: 85179
O-Subject: [RHEL7.1 PATCH] be2net: enable interrupts in EEH resume
Bugzilla: 1076682
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Stefan Assmann <sassmann@redhat.com>
RH-Acked-by: Steve Best <sbest@redhat.com>

BZ: #1076682 - be2net: EEH recovery fail in Rains adapter

Description:
The patch fixes a problem with failure during EEH recovery on BE3 chipsets.

Test status:
Tested successfully by IBM

Upstream commit(s):
commit 03a58baa785f48a85126ab043a14cb80b7e670e0
Author: Kalesh AP <kalesh.purayil@emulex.com>
Date:   Tue May 13 14:03:11 2014 +0530

    be2net: enable interrupts in EEH resume

    On some BE3 FW versions, after a HW reset, interrupts will remain disabled
    for each function. So, explicitly enable the interrupts in the eeh_resume
    handler, else after an eeh recovery interrupts wouldn't work.

    Signed-off-by: Kalesh AP <kalesh.purayil@emulex.com>
    Signed-off-by: Sathya Perla <sathya.perla@emulex.com>
    Signed-off-by: David S. Miller <davem@davemloft.net>

Signed-off-by: Ivan Vecera <ivecera@redhat.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/emulex/benet/be_main.c b/drivers/net/ethernet/emulex/benet/be_main.c
index 154e19e..8e0e106 100644
--- a/drivers/net/ethernet/emulex/benet/be_main.c
+++ b/drivers/net/ethernet/emulex/benet/be_main.c
@@ -4725,6 +4725,12 @@ static void be_eeh_resume(struct pci_dev *pdev)
  if (status)
   goto err;
 
+ /* On some BE3 FW versions, after a HW reset,
+  * interrupts will remain disabled for each function.
+  * So, explicitly enable interrupts
+  */
+ be_intr_set(adapter, true);
+
  /* tell fw we're ready to fire cmds */
  status = be_cmd_fw_init(adapter);
  if (status)
-- 
1.7.1